### PR TITLE
Feature/server/project file refactoring

### LINKED ISSUE #104: Flask: Return json instead of html on error
Currently Flask returns html documents when an error occurs. This should be changed so that json documents are returned with information about the error.

### PR DESCRIPTION
**Pull request by YIllmann**
This PR refactors the deprecated web module and moves the endpoints to their related modules.

In Detail:
- `/login` and `/csrf` moved to `administration`
- `/templates` moved to `scenario`
- rest moved to `execution web_api`

Additionally the defined error handling is globally defined on the flask app and extended by a general json return of error for every client and server error. Further the `flask.py` in `utils` was redefined and to decorator. Therefore the method `try_get_execution` was moved into `execution.utils`.

Every `api` endpoint registration of execution is relocated into a separate setup file.

Due to missing login-documentation of the web frontend, I was not able to validate the endpoints functionality. 

Closes #104

### CODE REVIEW

File: server/administration/web_api/login.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
+from flask import Blueprint
+from flask_api import status
+from flask_jwt_extended import create_access_token
+from flask_login import login_user
+
+import models
+from utils.decorator import required, RequiredValueSource
+
+web_api = Blueprint("web_api-scenario", __name__)
```
Comment (H1ghBre4k3r) [inline server/administration/web_api/login.py:9]: Shouldn't this be `web_api-login`?

File: server/administration/web_api/security.py [hunk 2]
```diff
@@ -0,0 +1,12 @@
+from flask import Blueprint
+from flask_wtf.csrf import generate_csrf
+
+from app_config import csrf
+
+web_api = Blueprint("web_api-scenario", __name__)
```
Comment (H1ghBre4k3r) [inline server/administration/web_api/security.py:6]: Shouldn't this be `web_api-security`?

File: server/app.py [hunk 3]
```diff
@@ -36,18 +36,23 @@ def create_app(csrf: CSRFProtect, db: SQLAlchemy):
...
     csrf.init_app(app)
     jwt = JWTManager(app)
-    web.setup(app)      # FIXME deprecated package
-    execution.web.setup.setup(app)
+
+    # -- Endpoint connection
+    execution.web_api.setup.setup(app)
+    scenario.web_api.setup.setup(app)
+    execution.api.setup.setup(app)
+
```
Comment (H1ghBre4k3r) [inline server/app.py:44]: Here, we are missing the registration for the administration API.

File: server/app.py [hunk 4]
```diff
@@ -36,18 +36,23 @@ def create_app(csrf: CSRFProtect, db: SQLAlchemy):
...
     CORS(app, resources={r"/*": {"origins": "*"}})
 
-    # define run request blocker
+    # -- Pre Request Handler
     @app.before_request
     def return_hold_if_not_running():
+        """ Blocks all run-request on an execution which is still in status pending. """
```
Comment (H1ghBre4k3r) [inline server/app.py:50]: I think, we should think about the semantics of this function. Currently, it does not block anything. However, if an execution is _not running_, we simply return `204` (which, additionally, means `No Content`, but we return a message - thus not following the HTTP semantics). 

If this function returns nothing, we just execute the regular request handlers.

**Review by H1ghBre4k3r — CHANGES_REQUESTED** [review]

File: server/app.py (hunk 4 above)
Comment (YIllmann) [reply server/app.py:52]: This function detects if an execution is not set to status running. If so, the function simply returns an empty message. Therefore it blocks any state changing requests on the api. afaik this method is tested and proven. However we can argue about the location of the request handler. We can relocate it to the registered api endpoints only and exclude the web-api.

**Review by H1ghBre4k3r — APPROVED** [review]

**Review by simohlsen — APPROVED** [review]

**Pull request merged**